src/play_scraper.py

```diff
@@ -0,0 +1,56 @@
+from urllib.request import urlopen, urlretrieve
+from pathlib import Path
+from bs4 import BeautifulSoup
+
+
+def _get_html(url: str) -> BeautifulSoup:
+    page = urlopen(url)
+    soup = BeautifulSoup(page, 'html.parser')
+    return soup
+
+
+class PlayPageScraper:
+    """A class for scraping Google Play Store web pages."""
+
+    _ICON_CLASS = "T75of sHb2Xb"  # icon's tag's class
+    _CATEGORY_ITEMPROP = "genre"  # category's tag's itemprop
+
+    def __init__(self, base_url: str, storage_dir: Path):
+        """Constructor.
+
+        :param base_url: base url of the apps' web pages.
+        :param storage_dir: main storage directory for retrieved info.
+        """
+        self._base_url = base_url
+
+        self._storage_dir = storage_dir
+        self._storage_dir.mkdir(exist_ok=True, parents=True)
+
+    def get_icon(self, app_id: str, directory: Path = "") -> None:
```

Comment:
this method is quite involved, let's split it into:
-> get_app_details_page(app_id) (returns the app details page html contents)
-> scrape_app_icon_url(html) (scrapes the app icon url given a page)
-> download_app_icon(icon_url, directory) (creates the directory and downloads&saves the image there)
-> download_app_icon(app_id, directory) (same as your original method)

By splitting the methods you'd be able to easily test individual logic (and corner cases like failed requests), improve readability, and more clearly handle failures due to different methods returning unexpected results.

Reply:
I see that the last two suggested functions have the same name. Can you please elaborate on it? Do you want to overload this method or is it just a mistake?

Reply:
Whoops, sorry about the confusion, as per our discussion I meant the download_app_icon(icon_url, directory) to be download_img(img_url, directory)